src/huggingface_hub/_snapshot_download.py

```diff
@@ -158,26 +158,20 @@ def snapshot_download(
     # If the specified revision is a commit hash, look inside "snapshots".
     # If the specified revision is a branch or tag, look inside "refs".
     if local_files_only:
-        commit_hash = revision
-        if not REGEX_COMMIT_HASH.match(commit_hash):
-            # rertieve commit_hash from file
+
+        def resolve_ref(revision) -> str:
+            # retrieve commit_hash from file
             ref_path = os.path.join(storage_folder, "refs", revision)
             with open(ref_path) as f:
-                commit_hash = f.read()
+                return f.read()
 
+        commit_hash = (
+            revision if REGEX_COMMIT_HASH.match(revision) else resolve_ref(revision)
+        )
         snapshot_folder = os.path.join(storage_folder, "snapshots", commit_hash)
+
```

Comment:
See side by side comparison of before/after @julien-c:


<table>
<tr>
<td>After</td>
<td>Before</td>
</tr>

<tr>
<td>

```py
if local_files_only:

    def resolve_ref(revision) -> str:
        # retrieve commit_hash from file
        ref_path = os.path.join(storage_folder, "refs", revision)
        with open(ref_path) as f:
            return f.read()

    commit_hash = (
        revision if REGEX_COMMIT_HASH.match(revision) else resolve_ref(revision)
    )
    snapshot_folder = os.path.join(storage_folder, "snapshots", commit_hash)

    if os.path.exists(snapshot_folder):
        return snapshot_folder

    raise ValueError(
        "Cannot find an appropriate cached snapshot folder for the specified"
        " revision on the local disk and outgoing traffic has been disabled. To"
        " enable repo look-ups and downloads online, set 'local_files_only' to"
        " False."
    )
```

</td>
<td>

```py
if local_files_only:
    if REGEX_COMMIT_HASH.match(revision):
        snapshot_folder = os.path.join(storage_folder, "snapshots", revision)
        if os.path.exists(snapshot_folder):
            return snapshot_folder
    else:
        ref_path = os.path.join(storage_folder, "refs", revision)
        with open(ref_path) as f:
            commit_hash = f.read()
        snapshot_folder = os.path.join(storage_folder, "snapshots", commit_hash)
        if os.path.exists(snapshot_folder):
            return snapshot_folder
     raise ValueError(
        "Cannot find an appropriate cached folder for the specified revision on the"
        " local disk and outgoing traffic has been disabled. To enable repo"
        " look-ups and downloads online, set 'local_files_only' to False."
    )
```

</td>
</tr>

</table>

Reply:
Yep, looks great to me!